tests/basic_one_way_send.go

```diff
@@ -0,0 +1,175 @@
+		ctx := context.Background()
+
+		// Get the latest block from Subnet A, and retrieve the warp message from the logs
+		log.Info("Waiting for new block confirmation")
+		newHeadA := <-NewHeadsA
+		blockHashA := newHeadA.Hash()
+
+		log.Info("Fetching relevant warp logs from the newly produced block")
+		logs, err := ChainARPCClient.FilterLogs(ctx, interfaces.FilterQuery{
+			BlockHash: &blockHashA,
+			Addresses: []common.Address{warp.Module.Address},
+		})
+		Expect(err).Should(BeNil())
+		Expect(len(logs)).Should(Equal(1))
+
+		// Check for relevant warp log from subscription and ensure that it matches
+		// the log extracted from the last block.
+		txLog := logs[0]
+		log.Info("Parsing logData as unsigned warp message")
+		unsignedMsg, err := avalancheWarp.ParseUnsignedMessage(txLog.Data)
+		Expect(err).Should(BeNil())
+
+		// Set local variables for the duration of the test
+		unsignedWarpMessageID := unsignedMsg.ID()
+		unsignedWarpMsg := unsignedMsg
+		log.Info("Parsed unsignedWarpMsg", "unsignedWarpMessageID", unsignedWarpMessageID, "unsignedWarpMessage", unsignedWarpMsg)
+
+		// Loop over each client on chain A to ensure they all have time to accept the block.
+		// Note: if we did not confirm this here, the next stage could be racy since it assumes every node
+		// has accepted the block.
+		WaitForAllValidatorsToAcceptBlock(ctx, ChainANodeURIs, BlockchainIDA, newHeadA.Number.Uint64())
+
+		// Get the aggregate signature for the Warp message
+		log.Info("Fetching aggregate signature from the source chain validators")
+		warpClient, err := warpBackend.NewClient(ChainANodeURIs[0], BlockchainIDA.String())
+		Expect(err).Should(BeNil())
+		signedWarpMessageBytes, err := warpClient.GetAggregateSignature(ctx, unsignedWarpMessageID, params.WarpQuorumDenominator)
+		Expect(err).Should(BeNil())
+
+		// Construct the transaction to send the Warp message to the destination chain
+		signedTxB := ConstructAndSendTransaction(
+			ctx,
+			signedWarpMessageBytes,
+			teleporterMessage,
+			TeleporterContractAddress,
+			FundedAddress,
+			FundedKey,
+			ChainBRPCClient,
+			ChainBIDInt,
+		)
+
+		// Sleep to ensure the new block is published to the subscriber
+		time.Sleep(5 * time.Second)
+		receipt, err := ChainBRPCClient.TransactionReceipt(ctx, signedTxB.Hash())
+		Expect(err).Should(BeNil())
+		Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+
+		sendCrossChainMessageLog := receipt.Logs[0]
+		var event SendCrossChainMessageEvent
+		err = teleporter.EVMTeleporterContractABI.UnpackIntoInterface(&event, "SendCrossChainMessage", sendCrossChainMessageLog.Data)
+		Expect(err).Should(BeNil())
+		teleporterMessageID = event.Message.MessageID
+	})
+
+	ginkgo.It("Check Teleporter Message Received", ginkgo.Label("Teleporter", "TeleporterMessageReceived"), func() {
+		time.Sleep(5 * time.Second) // Give the relayer a chance to deliver the message to the destination chain
```

Comment:
I think this waiting is redundant given that we've already waited in the `It` above for a new block to be produced after sending the transaction on the destination, right?

Reply:
This sleep is to give the relayer a chance to construct the aggregate signatures and deliver to the destination. We could probably replace this wait with a `SubscriberNewHead` on the destination chain. Geoff made a similar suggestion, so I'll look into implementing this.

Reply:
Removed.